docs/database-classes.md

```diff
@@ -1,348 +1,283 @@
 {
-    public IEnumerable<Person> SelectShareHolders(IDatabaseContext db)
+    public abstract string Name { get; set; }
+    public abstract Person Ceo { get; set; }
+
+    public IEnumerable<Person> Shareholders
     {
-        return db.Sql<Person>
-        (
-            "SELECT s.Owner FROM Shares s WHERE s.Equity = ?",
-            this
-        );
+        get
+        {
+            var db = DbProxy.GetContext(this);
+            return db.Sql<Person>("SELECT s.Owner FROM HoldsShares s WHERE s.Equity = ?", this);
+        }
     }
 }
 
 [Database]
-public class Shares
+public abstract class HoldsShares
 {
-    public virtual Person Owner { get; set; }
-    public virtual Company Equity { get; set; }
-    public virtual int Quantity { get; set; }
+    [Index]
+    public abstract Person Owner { get; set; }
+
+    [Index]
+    public abstract Company Equity { get; set; }
+
+    public abstract int Quantity { get; set; }
 }
 ```
 
 ### Inheritance
 
-Any database class can inherit from any other database class.
+Inheritance is supported between database classes and has the same general semantics as class inheritence in C#.
 
 ```csharp
 using Starcounter.Database;
 
 [Database]
-public class Customer
+public abstract class Customer
 {
-   public virtual string Name { get; set; }
+   public abstract string Name { get; set; }
 }
 
-public class PrivateCustomer : Customer
+public abstract class PrivateCustomer : Customer
 {
-   public virtual string Gender { get; set; }
+   public abstract string Gender { get; set; }
 }
 
-public class CorporateCustomer : Customer
+public abstract class CorporateCustomer : Customer
 {
-   public virtual string VatNumber { get; set; }
+   public abstract string VatNumber { get; set; }
 }
 ```
 
-The `[Database]` attribute is inherited from base - to subclasses. Any class that directly or indirectly inherits a class with the `[Database]` attribute becomes a database class. In the example above, both `PrivateCustomer` and `CorporateCustomer` become database classes due to them inheriting `Customer`.
-
-The table `Customer` will contain all `PrivateCustomers` and all `CorporateCustomers`. So if there is a private customer called `"Goldman, Carl"` and a corporate customer called `"Goldman Sachs"`, the result of `SELECT C FROM Customer c` will contain both of them.
+The `DatabaseAttribute` attribute decoration is inherited from any base class to its subclasses, meaning that any class that directly or indirectly inherits a class that is decorated with `DatabaseAttribute` becomes a database class. In the example above, both `PrivateCustomer` and `CorporateCustomer` become database classes due to them inheriting `Customer`.
 
-#### Inheriting from non-database classes
+This also means that all rows in the `CorporateCustomer` table also are contained in the `Customer` table. The result of the `SELECT C FROM Customer c` SQL query will contain all rows from `Customer` as well as all tables from subclasses of `Customer`.
```

Comment:
```suggestion
This also means that all rows in the `CorporateCustomer` table also are contained in the `Customer` table. The result of the `SELECT C FROM Customer c` SQL query will contain all rows from `Customer` as well as all rows from sub-classes of `Customer`.
```

Reply:
Thanks for spotting the "tables" typo. Regarding "subclasses", it's "[subclass](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)#Subclasses_and_superclasses)", not "sub-class"